### ISSUE TITLE
Contributing guide - getting started

### ISSUE THREAD

**Issue by stephenlautier** [dotnet/orleans#3971]
I and my team are willing to try and contribute to Orleans source ,however, currently there's no guide on how to start and for a big project like this we don't know where to start.

Can you create a very simple guide on how to get started in order to contribute e.g. clone, build, run (ideally in our project so we can test better + debuggable)

Something along these lines is what im thinking:

In order to contribute to the project follow these simple steps:

- Clone orleans repo - `git clone https://github.com/dotnet/orleans.git`
- Restore/Build project - `dotnet build .\Orleans.sln`
- Reference Orleans csproj instead the nugets e.g. `Orleans.Core.csproj` in your app (example)
- ...
- Run tests via `Test.cmd`

If you are able to help us where to start, we are willing to contribute, thanks!

**Comment by ReubenBond** [comment]
Good idea, @stephenlautier :) Maybe we should work out the various steps required here on this issue and then compile them into a short document.

* To build, use the `Build.cmd` script in the repo root.
* If you need code generation in your projects (quite likely) then you need to modify your csproj to include:
```xml
<PropertyGroup>
  <OrleansBuildTimeCodeGen>true</OrleansBuildTimeCodeGen>
</PropertyGroup>
```
* Projects need to be under the same tree as Orleans because it is important that Orleans' `Directory.Build.props` and `Directory.Build.targets` files are in a parent directory of the project.

**Comment by sergeybykov** [comment]
Good idea, indeed! I see that we have some pieces on https://github.com/dotnet/orleans/blob/master/README.md and some other on http://dotnet.github.io/orleans/Community/Contributing.html. I wonder if we want to move this info to http://dotnet.github.io/orleans/Documentation/Introduction.html instead and add a link to it to https://github.com/dotnet/orleans/blob/master/README.md?
